File: JavaScript/searching/binary_search.js
const binarySearch = (arr, target) => {
  let left = 0;
  let right = arr.length - 1;
  while (left <= right) {
    midpoint = Math.floor((left + right) / 2);
    if (arr[midpoint] === target) {
      return midpoint;
    } else if (arr[midpoint] < target) {
      left = midpoint + 1;
    } else {
      right = midpoint - 1;
    }
  }
  return -1;
};

const myArray = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19];
const myTarget = 7;
const result = binarySearch(myArray, myTarget);

if (result !== -1) {
  console.log(`Value ${myTarget} found at index ${result}`);
} else {
  console.log("Target not found in array.");
}

File: searching algorithms/binary search/binary_search.py
def binarySearch(arr, target):
    left = 0
    right = len(arr) - 1
    while left <= right:
        mid = (left + right) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return -1


myArray = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19]
myTarget = 3
result = binarySearch(myArray, myTarget)


if result != -1:
    print("Value", myTarget, "found at index", result)
else:
    print("Target not found in array.")
